### ISSUE TITLE
Cache the Document title

### ISSUE THREAD

**Issue by nox** [servo/servo#6713]
Whenever Document.title_changed() is called by the virtual methods on HTMLTitleElement, the tree is traversed to find the first title element in the document and notify the chrome of the change.

The current title element should be cached in Document in a MutNullableHeap&lt;JS&lt;HTMLTitleElement&gt;&gt; field and be updated in bind_from_tree() and unbind_from_tree() in the virtual methods on HTMLTitleElement.

**Comment by nox** [comment]
Cc'ing @nick-thompson who said he was interested in looking into this.

**Comment by nick-thompson** [comment]
Thanks, I'm on it!

**Comment by frewsxcv** [comment]
> ...and be updated in bind_from_tree() and unbind_from_tree() in the virtual methods on HTMLTitleElement

Sinmce an SVG `<title>` element is (presumably) not an `HTMLTitleElement`, how will this work in the SVG case?

**Comment by elomscansio** [comment]
can i look into this

**Comment by jdm** [comment]
Yes, that sounds fine. The discussion in #6744 looks like the implementation was acceptable.

**Comment by elomscansio** [comment]
I have a [PR](https://github.com/servo/servo/pull/36742)
